lib/dotiw/methods.rb

```diff
@@ -115,7 +152,52 @@ def _display_time_in_words(hash, options = {})
                                                        default: :'support.array.last_word_connector',
                                                        locale: options[:locale]
 
-      output.to_sentence(options.except(:accumulate_on, :compact))
+      phrases.to_sentence(options.except(:accumulate_on, :compact))
+    end
+
+    def _compute_highest_measures!(options)
+      highest_measures = options.delete(:highest_measures)
+      highest_measures = 1 if options.delete(:highest_measure_only)
+      highest_measures = { max: highest_measures } if highest_measures.is_a?(Integer)
+      highest_measures = highest_measures.reverse_merge(max: 1, remainder: :floor) if highest_measures
+
+      highest_measures
+    end
+
+    def _maybe_round!(hash, discarded_hash, remainder)
+      smallest_measure_index = DOTIW::TimeHash::TIME_FRACTIONS.index hash.to_a.last[0]
+      smallest_measure = DOTIW::TimeHash::TIME_FRACTIONS[smallest_measure_index]
+
+      case remainder
+      when :ceiling
+        # We already filtered out zeroes, so non-empty also means non-zero.
+        if !discarded_hash.empty?
+          hash[smallest_measure] += 1
+          _rollup! hash, smallest_measure_index
+        end
+      when :round
+        # If our smallest measure is already the smallest possible measure, there is no next
+        # smallest measure to inspect to see if we need to round up.
+        return if smallest_measure_index == 0
+
+        next_smallest_measure = DOTIW::TimeHash::TIME_FRACTIONS[smallest_measure_index - 1]
+        if discarded_hash.fetch(next_smallest_measure, 0) >= ROUNDING_THRESHOLDS[next_smallest_measure]
+          hash[smallest_measure] += 1
+          _rollup! hash, smallest_measure_index
+        end
```

Comment:
You're actually not checking for `:floor`, so `:anything` will also work. Explicitly allow `:floor` and raise an error when it's something unexpected + tests.

Reply:
Sure.